src/Storage/Builder.php

```diff
@@ -0,0 +1,189 @@
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Storage;
+
+use Lcobucci\Jose\Parsing;
+use Lcobucci\JWT\Builder as BuilderInterface;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Signer\Key;
+use Lcobucci\JWT\Token as TokenInterface;
+
+/**
+ * This class makes easier the token creation process
+ *
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ * @since 0.1.0
+ */
+final class Builder implements BuilderInterface
+{
+    /**
+     * The token header
+     *
+     * @var array
+     */
+    private $headers = ['typ'=> 'JWT', 'alg' => 'none'];
+
+    /**
+     * The token claim set
+     *
+     * @var array
+     */
+    private $claims = [];
+
+    /**
+     * The data encoder
+     *
+     * @var Parsing\Encoder
+     */
+    private $encoder;
+
+    /**
+     * Initializes a new builder
+     */
+    public function __construct(Parsing\Encoder $encoder)
+    {
+        $this->encoder = $encoder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canOnlyBeUsedBy(string $audience, bool $addHeader = false): BuilderInterface
+    {
+        $audiences = $this->claims['aud'] ?? [];
+
+        if (!in_array($audience, $audiences)) {
+            $audiences[] = $audience;
+        }
+
+        return $this->setRegisteredClaim('aud', $audiences, $addHeader);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function expiresAt(int $expiration, bool $addHeader = false): BuilderInterface
+    {
+        return $this->setRegisteredClaim('exp', $expiration, $addHeader);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function identifiedBy(string $id, bool $addHeader = false): BuilderInterface
+    {
+        return $this->setRegisteredClaim('jti', $id, $addHeader);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function issuedAt(int $issuedAt, bool $addHeader = false): BuilderInterface
+    {
+        return $this->setRegisteredClaim('iat', (int) $issuedAt, $addHeader);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function issuedBy(string $issuer, bool $addHeader = false): BuilderInterface
+    {
+        return $this->setRegisteredClaim('iss', $issuer, $addHeader);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canOnlyBeUsedAfter(int $notBefore, bool $addHeader = false): BuilderInterface
+    {
+        return $this->setRegisteredClaim('nbf', $notBefore, $addHeader);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function relatedTo(string $subject, bool $addHeader = false): BuilderInterface
+    {
+        return $this->setRegisteredClaim('sub', $subject, $addHeader);
+    }
+
+    /**
+     * Configures a registered claim
+     */
+    private function setRegisteredClaim(string $name, $value, bool $addHeader): BuilderInterface
+    {
+        $this->with($name, $value);
+
+        if ($addHeader) {
+            $this->headers[$name] = $this->claims[$name];
```

Comment:
Although they're private properties and _you know_ what `with()` does, I would create some method to get the claims by name, that you could also use on [`canOnlyBeUsedBy()`](https://github.com/lcobucci/jwt/pull/129/files#diff-343ce2f1a7cebf5e8711bc1f726796abR60), and I would also use `withHeader()` instead of setting `$this->headers` manually.